src/Microsoft.Health.Dicom.Api/Features/Exceptions/ExceptionHandlingMiddleware.cs

```diff
@@ -160,6 +172,12 @@ private static bool IsTaskCanceledException(Exception ex)
         return ex is TaskCanceledException || (ex is AggregateException aggEx && aggEx.InnerExceptions.Any(x => x is TaskCanceledException));
     }
 
+    private static bool IsCMKException(Exception ex)
+    {
+        return ex is SqlException sqlEx && sqlEx.Number is SQLKeyVaultCriticalError or SQLKeyVaultEncounteredError or SQLKeyVaultErrorObtainingInfo or SQLCannotConnectToDBInCurrentState ||
```

Comment:
Most of our SQL Exception are transformed to DataStoreException. May be you need to do 

```
var exception = ex is SqlException ? ex : (ex.InnerException is SqlException ? ex.InnerException : null)
if(exception != null) 
{
return exception .Number is SQLKeyVaultCriticalError or SQLKeyVaultEncounteredError or SQLKeyVaultErrorObtainingInfo or SQLCannotConnectToDBInCurrentState
}
```

Reply:
And we have missed in some places, I have a PR to add a try catch in all the GET calls. #3239

Reply:
Yes, but I believe this is enough and we can improve it later if we find it